### PR TITLE
AP-4140: Merge vehicle questions into single page

### PR DESCRIPTION
**Pull request by colinbruce**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4140)

- [x] Create a new form to handle all questions
- [x] Create a new controller to handle the combined page
- [x] Refactor the vehicle flow with the new path
- [x] Update features to expect and use combined page
- [x] Remove the original controller, forms, routes and tests
- [x] Create a rake task to migrate users on the original pages to the new one

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/providers/means/vehicle_details/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<%= form_with(
```
Comment (naseberry) [inline app/views/providers/means/vehicle_details/show.html.erb:1]: Validation: to endure error summary correctly links to the first radio button, do we need set the option `link_errors` to `true`?

**Review by jsugarman — COMMENTED** [review]
Looks good.

Mainly minor, ignorable, nits but some spec variable values, payment_remains and payment_remaining look mixed up, unless i am going blind; An update suggestion unless callbacks/validations are needed;

File: app/views/providers/means/vehicle_details/show.html.erb (hunk 1 above)
Comment (colinbruce) [reply app/views/providers/means/vehicle_details/show.html.erb:1]: Only the payments_remain needed one.  The govuk helpers handle the others, e.g. govuk_text_field links automatically :+1:

File: features/support/steps_helper.rb [hunk 2]
```diff
@@ -114,6 +114,11 @@ def run_axe?
   choose(field.parameterize(separator: "-"), allow_label_click: true)
 end
 
+Then("I answer/change {string} with/to {string}") do |question, answer|
+  control = find(:xpath, "//h2[text()='#{question}']/ancestor::fieldset[1]//child::label[contains(text(), '#{answer}')]")
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply features/support/steps_helper.rb:118]: If you can find a method of finding a specific `Yes` button on page without xpathing from the qustion I'd love some pointers! 😄

File: app/forms/vehicles/details_form.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                  :more_than_three_years_old,
+                  :payment_remaining,
+                  :payments_remain,
+                  :used_regularly
+
+    validates :estimated_value,
+              currency: { greater_than_or_equal_to: 0, allow_blank: true },
+              presence: { unless: :draft? }
+    validates :more_than_three_years_old, presence: { unless: :draft? }
+    validates :payments_remain, presence: true, unless: :draft?
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply app/forms/vehicles/details_form.rb:15]: Good shout, these were merged from disparate forms that used different patterns!
Also, I have re-sequences the validation as the error list did not match the sequence of questions :+1:

File: spec/requests/providers/means/vehicle_details_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+      context "and checking non passported answers" do
+        let(:legal_aid_application) { create(:legal_aid_application, :with_non_passported_state_machine, :checking_non_passported_means) }
+
+        it "redirects to check capital answers page" do
+          patch_vehicle_details
+          expect(response).to redirect_to(providers_legal_aid_application_check_capital_answers_path(legal_aid_application))
+        end
+      end
+
+      context "and checking passported answers" do
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply spec/requests/providers/means/vehicle_details_controller_spec.rb:129]: Ditto above for nesting and line 130 is 

```
       let(:legal_aid_application) { create(:legal_aid_application, :with_passported_state_machine, :checking_passported_answers) }
``` 
so changing to non passported wouldn't be right!

File: lib/tasks/migrate_vehicle_provider_steps.rake [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    Rails.logger.info "Vehicles ages page count            : #{applications.where(provider_step: 'vehicles_ages').count}"
+    Rails.logger.info "Vehicles estimated values page count: #{applications.where(provider_step: 'vehicles_estimated_values').count}"
+    Rails.logger.info "Vehicles remaining payments count   : #{applications.where(provider_step: 'vehicles_remaining_payments').count}"
+    Rails.logger.info "Vehicles regular use count          : #{applications.where(provider_step: 'vehicles_regular_uses').count}"
+    Rails.logger.info "-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+    Benchmark.benchmark do |bm|
+      bm.report("Migrate:") do
+        ActiveRecord::Base.transaction do
+          applications.each do |app|
+            app.provider_step = "vehicle_details"
```
[the remark this answers is not in the document]
Comment (colinbruce) [reply lib/tasks/migrate_vehicle_provider_steps.rake:18]: There are less that a hundred and it takes less than 2 seconds! :)

File: features/support/steps_helper.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (naseberry) [reply features/support/steps_helper.rb:118]: I think we (i.e Joel) did something similar on CCCD that was easier to read, although still using `xpath`

```
Then("I answer/change {string} with/to {string}") do |question, answer|
  find(".govuk-fieldset__legend", text: question)
    .find(:xpath, "..")
    .find(".govuk-radios__item label", text: answer).click
end
```

[the remark this answers is not in the document]
Comment (colinbruce) [reply features/support/steps_helper.rb:118]: How do we feel about complete removal of xpath, with
```ruby
  find(".govuk-fieldset__legend", text: question)
    .sibling(".govuk-radios")
    .find(".govuk-radios__item label", text: answer).click
```

[the remark this answers is not in the document]
Comment (jsugarman) [reply features/support/steps_helper.rb]: I like it, certainly clearer 🎉

**Review by jsugarman — APPROVED** [review]
LGTM 👍 

There is one nit comment you 👍 but does not looked like it has been changed - using `£` in the params for testing cleaning. but either way good to go/UAT.

**Pull request merged**